documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+
+In the v1 SDK, the [awsIot.device](https://github.com/aws/aws-iot-device-sdk-js?tab=readme-ov-file#device) module represents
+an MQTT client. You instantiate the client directly by passing all the required parameters to the class constructor. It's
+possible to change the client settings after its creation using the `set*` methods like `setKeepAliveInterval` or `setMaxConnectionRetries`.
+
+In the v2 SDK, the [Mqtt5Client](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/mqtt5.Mqtt5Client.html) class
+represents an MQTT client, specifically for MQTT5 protocol. The v2 SDK provides an [MQTT5 client builder](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/iot.AwsIotMqtt5ClientConfigBuilder.html)
+designed to easily create common configuration types such as direct MQTT or WebSocket connections. After an MQTT5 client
+is built and finalized, the settings of the resulting MQTT5 client cannot be modified.
+
+#### Example of creating a client in the v1 SDK
+
+```typescript
+var awsIot = require('aws-iot-device-sdk');
+
+var clientEndpoint = "<prefix>-ats.iot.<region>.amazonaws.com";
+var clientId = "<unique client id>";
+var certificateFile = "<certificate file>";  // X.509 based certificate file
+var privateKeyFile = "<private key file>";   // PEM encoded private key file
+
+var device = awsIot.device({
+        keyPath: privateKeyFile,
+        certPath: certificateFile,
+        clientId: clientId,
+        host: clientEndpoint
+});
+```
+
+#### Example of creating a client in the v2 SDK
+
+The v2 SDK supports different connection types. Given the same input parameters as in the v1 example above, the recommended
+method to create an MQTT5 client will be [newDirectMqttBuilderWithMtlsFromPath](https://aws.github.io/aws-iot-device-sdk-js-v2/node/classes/iot.AwsIotMqtt5ClientConfigBuilder.html#newDirectMqttBuilderWithMtlsFromPath). 
+
+```typescript
+import { mqtt5, iot } from "aws-iot-device-sdk-v2";
+
+const clientEndpoint : string = "<prefix>-ats.iot.<region>.amazonaws.com";
+const clientId : string = "<unique client id>";
+const certificateFile : string = "<certificate file>";  // X.509 based certificate file
+const privateKeyFile : string = "<private key file>";   // PEM encoded private key file
+
+let builder = iot.AwsIotMqtt5ClientConfigBuilder.newDirectMqttBuilderWithMtlsFromPath(
+        clientEndpoint,
+        certificateFile,
+        privateKeyFile
+
+builder.withConnectProperties({
+    clientId: clientId
+});
+
+let config : mqtt5.Mqtt5ClientConfig = builder.build();
+
+let client : mqtt5.Mqtt5Client = new mqtt5.Mqtt5Client(config);
+```
+
+For more information, refer to the [Connection types and features](#connection-types-and-features) section for other connection types supported by the v2 SDK.
+
+### Client start
+
+To connect to the server in the v1 SDK, you just instantiate a `device` object.
+
+In the v2 SDK, you explicitly `start` the MQTT5 client.
```

Comment:
Trivial: Above we state instantiating a device object connects. Instead of "explicitly `start` the MQTT5 client." maybe we can state what we're doing. e.g. "To connect to the server in the v2 SDK, you call the `start()` function of the MQTT5 client."